### PR TITLE
fix OpenAI embedding spec for batching

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?

This PR addresses two key issues related to request decoding and response formatting when using the OpenAIEmbeddingSpec:

**1. Request Input Wrapping:**
The decode_request function always wraps request.input in a list, which breaks the downstream logic for certain input formats. This PR ensures inputs are handled more intelligently based on their structure.

**2. Output Shape Handling:**
When predict returns a 1D tensor or NumPy array (e.g., shape (128,)), the OpenAIEmbeddingSpec attempts to wrap each element individually into the Embedding class, leading to incorrect behavior. This PR corrects that by ensuring the output is treated as a single embedding vector when appropriate.

✅ Example Use Case
The following example previously failed due to the issues above but now works as expected after this fix:

```python
from sentence_transformers import SentenceTransformer
import litserve as ls

class EmbeddingsAPI(ls.LitAPI):
    def setup(self, device):
        self.model = SentenceTransformer("all-MiniLM-L6-v2", device=device)

    def predict(self, inputs):
        # Fix for nested list inputs
        if isinstance(inputs[0], list) and len(inputs[0]) == 1:
            inputs = inputs[0]
        return self.model.encode(inputs)

if __name__ == "__main__":
    api = EmbeddingsAPI(max_batch_size=2, batch_timeout=2)
    server = ls.LitServer(api, spec=ls.OpenAIEmbeddingSpec())
    server.run(port=8000)
```

After this PR, users will be able to run embedding models with LitServe and the OpenAIEmbeddingSpec seamlessly, even when the model outputs a 1D vector or the input structure is slightly irregular.

<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/litserve/specs/base.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def __init__(self):
         self._endpoints = []
 
         self._server: LitServer = None
+        self._max_batch_size = 1
```
Comment (KaelanDt) [inline src/litserve/specs/base.py:28]: is there a reason to hard code the batch size to 1?

File: tests/test_openai_embedding.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    with wrap_litserve_start(server) as server:
+        async with LifespanManager(server.app) as manager, AsyncClient(
+            transport=ASGITransport(app=manager.app), base_url="http://test"
+        ) as ac:
+            resp = await ac.post("/v1/embeddings", json=openai_embedding_request_data, timeout=10)
+            assert resp.status_code == 200, "Status code should be 200"
+            assert resp.json()["object"] == "list", "Object should be list"
+            assert resp.json()["data"][0]["index"] == 0, "Index should be 0"
+            assert len(resp.json()["data"]) == 1, "Length of data should be 1"
+            assert len(resp.json()["data"][0]["embedding"]) == 768, "Embedding length should be 768"
```
Comment (KaelanDt) [inline tests/test_openai_embedding.py:87]: is there not a way to get the values here without hardcoding?

File: tests/test_specs.py [hunk 3]
```diff
@@ -337,128 +324,6 @@ async def test_fail_http(openai_request_data):
             assert res.text == '{"detail":"test LitAPI.predict error"}'
 
 
-@pytest.mark.asyncio
```
Comment (KaelanDt) [inline tests/test_specs.py:340]: what is the rationale for moving this test to another file?

**Review by KaelanDt — APPROVED** [review]
small nits, overall looks good to me!

File: src/litserve/specs/openai_embedding.py [hunk 4]
```diff
@@ -152,8 +188,60 @@ def _validate_response(self, response: dict) -> None:
...
+    def _handle_embedding_response(
+        self, embeddings: Union[List, "np.ndarray", "torch.Tensor", "List[List[float]]"], num_items: int = 1
+    ) -> List[Embedding]:
+        ndim = None
+        if "torch" in sys.modules:
+            import torch
+
+            if isinstance(embeddings, torch.Tensor):
+                ndim = embeddings.ndim
+        if "numpy" in sys.modules:
```
Comment (Borda) [inline src/litserve/specs/openai_embedding.py:200]:
```suggestion
        elif "numpy" in sys.modules:
```

File: src/litserve/specs/openai_embedding.py [hunk 5]
```diff
@@ -152,8 +188,60 @@ def _validate_response(self, response: dict) -> None:
...
+            import numpy as np
+
+            if isinstance(embeddings, np.ndarray):
+                ndim = embeddings.ndim
+
+        # expand_dims for torch.Tensor or np.ndarray
+        if ndim == 1:
+            embeddings = embeddings[None, :]
+
+        if ndim is not None:
```
Comment (Borda) [inline src/litserve/specs/openai_embedding.py:210]:
```suggestion
        elif ndim is not None:
```

File: src/litserve/specs/base.py (hunk 1 above)
Comment (aniketmaurya) [reply src/litserve/specs/base.py:28]: it's default to 1

File: tests/test_openai_embedding.py (hunk 2 above)
Comment (aniketmaurya) [reply tests/test_openai_embedding.py:87]: these are hardcoded since it's a test. We want to make sure it's 768 since the API implements 768 embedding dim and similarly for other items it should be static in every condition.

File: tests/test_specs.py (hunk 3 above)
Comment (aniketmaurya) [reply tests/test_specs.py:340]: moved OpenAIEmbedding based tests to test_openai_embedding.py

File: src/litserve/specs/openai_embedding.py (hunk 5 above)
Comment (aniketmaurya) [reply src/litserve/specs/openai_embedding.py:210]: we want to apply this independent of the above condition

File: src/litserve/specs/openai_embedding.py (hunk 4 above)
Comment (aniketmaurya) [reply src/litserve/specs/openai_embedding.py:200]: we want to check both. so it wont be 'else'

File: tests/test_openai_embedding.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (aniketmaurya) [reply tests/test_openai_embedding.py:87]: right. can move it later
